rllib/policy/sample_batch.py

```diff
@@ -367,17 +375,10 @@ def shuffle(self) -> "SampleBatch":
         # meaningless).
         permutation = np.random.permutation(self.count)
 
-        def _permutate_in_place(path, value):
-            curr = self
-            for i, p in enumerate(path):
-                if i == len(path) - 1:
-                    curr[p] = value[permutation]
-                # Translate into list (tuples are immutable).
-                if isinstance(curr[p], tuple):
-                    curr[p] = list(curr[p])
-                curr = curr[p]
-
-        tree.map_structure_with_path(_permutate_in_place, self)
+        self_as_dict = {k: v for k, v in self.items()}
```

Comment:
is this really necessary? self already looks like a dict.

Reply:
We cannot change in place as any tuples in the nested structures are not alterable. I had them converted to lists before (in the old mapping fn), but I didn't like this solution as it changes the actual nested struct (and this caused errors in another location). If we want to change the SampleBatch in place, this is still the cleanest and most concise solution, I think.

Bottom line: `tree.map_structure(lamdba ..., self)` will not work (not w/o a much more complex mapping fn).

Reply:
interesting 👌 
thanks for the explanation.